iModelCore/ECPresentation/Source/Content/ContentSpecificationsHandler.cpp

```diff
@@ -226,18 +256,85 @@ static ECSchemaHelper::RelationshipPathsResponse FindRelatedPropertyPaths(Conten
 /*---------------------------------------------------------------------------------**//**
 * @bsimethod
 +---------------+---------------+---------------+---------------+---------------+------*/
-bvector<std::unique_ptr<RelatedPropertySpecificationPaths>> ContentSpecificationsHandler::_GetRelatedPropertyPaths(RelatedPropertyPathsParams const& params) const
+static bvector<std::unique_ptr<FlattenedRelatedPropertiesSpecification>> CreateForSelectClassFromModifiers(ECClassCR selectClass, bvector<ContentModifierCP> const& modifiers)
     {
-    // first build a flat list of all related properties specifications
-    auto flatSpecs = FlattenedRelatedPropertiesSpecification::Create(params.GetRelatedPropertySpecs(), RelatedPropertiesSpecificationScopeInfo(params.GetScopeCategorySpecifications()));
-    for (ContentModifierCP modifier : GetContext().GetRulesPreprocessor().GetContentModifiers())
+    bvector<std::unique_ptr<FlattenedRelatedPropertiesSpecification>> specs;
+    for (ContentModifierCP modifier : modifiers)
         {
-        if (params.GetSourceClassInfo().GetSelectClass().GetClass().Is(modifier->GetSchemaName().c_str(), modifier->GetClassName().c_str()))
+        if (selectClass.Is(modifier->GetSchemaName().c_str(), modifier->GetClassName().c_str()))
             {
             DiagnosticsHelpers::ReportRule(*modifier);
-            ContainerHelpers::MovePush(flatSpecs, FlattenedRelatedPropertiesSpecification::Create(modifier->GetRelatedProperties(), RelatedPropertiesSpecificationScopeInfo(modifier->GetPropertyCategories())));
+            ContainerHelpers::MovePush(specs, FlattenedRelatedPropertiesSpecification::Create(modifier->GetRelatedProperties(), RelatedPropertiesSpecificationScopeInfo(modifier->GetPropertyCategories())));   
             }
         }
+    return specs;
+    }
+
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+static bvector<std::unique_ptr<FlattenedRelatedPropertiesSpecification>>CreateForNestedPropertiesFromModifiers(bvector<std::unique_ptr<FlattenedRelatedPropertiesSpecification>> const& flatSpecs, bvector<ContentModifierCP> const& modifiers, ECSchemaHelper const& helper);
+
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+static bvector<std::unique_ptr<FlattenedRelatedPropertiesSpecification>>CreateForFlatSpecFromModifiers(RelatedPropertiesSpecificationCR spec, bvector<ContentModifierCP> modifiers, ECSchemaHelper const& helper)

```

Comment:
missing space between return type and function name

Reply:
fixed